### PR TITLE
Use Option<T> over FFI

### PR DESCRIPTION
**Pull request by Manishearth**

### CODE REVIEW

File: ffi/capi/src/decimal.rs [hunk 1]
```diff
@@ -37,22 +38,14 @@ pub mod ffi {
...
-                FixedDecimalGroupingStrategy::Always => {
-                    icu_decimal::options::GroupingStrategy::Always
-                }
-                FixedDecimalGroupingStrategy::Min2 => icu_decimal::options::GroupingStrategy::Min2,
-            };
             let mut options = icu_decimal::options::FixedDecimalFormatterOptions::default();
-            options.grouping_strategy = grouping_strategy;
+            options.grouping_strategy = grouping_strategy
+                .map(Into::into)
+                .unwrap_or(icu_decimal::options::GroupingStrategy::Auto);
```
[the remark this answers is not in the document]
Comment (Manishearth) [reply ffi/capi/src/decimal.rs:48]: smart

**Comment by Manishearth** [comment]
Ready for review. I think this is the last large part of FFI future-proofing.

File: ffi/capi/src/casemap.rs [hunk 2]
```diff
@@ -300,8 +301,12 @@ impl From<ffi::TitlecaseOptionsV1> for TitlecaseOptions {
     fn from(other: ffi::TitlecaseOptionsV1) -> Self {
         let mut ret = Self::default();
 
-        ret.leading_adjustment = other.leading_adjustment.into();
-        ret.trailing_case = other.trailing_case.into();
+        if let Some(l) = other.leading_adjustment.into_converted_option() {
+            ret.leading_adjustment = l;
+        }
```
Comment (robertbastian) [inline ffi/capi/src/casemap.rs:306]: nit: use `unwrap_or` like in the other options conversions
```suggestion
        ret.leading_adjustment = other.leading_adjustment.into_converted_option().unwrap_or(ret.leading_adjustment);
```

File: ffi/capi/src/decimal.rs [hunk 3]
```diff
@@ -77,7 +70,7 @@ pub mod ffi {
             secondary_group_size: u8,
             min_group_size: u8,
             digits: &[DiplomatChar],
-            grouping_strategy: FixedDecimalGroupingStrategy,
+            grouping_strategy: Option<FixedDecimalGroupingStrategy>,
```
Comment (robertbastian) [inline ffi/capi/src/decimal.rs:73]: I think it's inconsistent to not have an options bag on some APIs, when the Rust API takes an options bag. Everywhere you call `Options::default()` in a constructor it should really take a bag.

Comment (Manishearth) [reply ffi/capi/src/decimal.rs:73]: Yeah, perhaps we should tweak this

**Review by robertbastian — APPROVED** [review]

**Pull request merged**